File: socialNetwork/src/main/java/com/ftn/socialNetwork/model/entity/GroupRequest.java
package com.ftn.socialNetwork.model.entity;

import javax.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Filter;
import org.hibernate.annotations.FilterDef;
import org.hibernate.annotations.ParamDef;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "groupRequests")
public class GroupRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private boolean approved ;

    @Column(nullable = false, columnDefinition = "DATETIME DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime created_at;

    @Column
    private LocalDateTime at;

    @ManyToOne
    private User user;
    @ManyToOne
    private Group group;
}

File: src/app/models/groupRequest.ts
import { Group } from "./group";
import { User } from "./user";

export class GroupRequest {
    id: number;
    approved: boolean ;
    created_at: Date;
    at: Date | null;
    user: User;
    group: Group;

  constructor(id: number,  approved: boolean ,created_at: Date, at: Date | null, user: User, group: Group
  ) { 
    this.id = id;
    this.approved = approved;
    this.created_at = created_at;
    this.at = at;
    this.user = user;
    this.group = group;

}

}
  